hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1128,3 +1129,44 @@ def hpat_pandas_series_le_impl(self, other):
         return hpat_pandas_series_le_impl
 
     raise TypingError('{} The object must be a pandas.series and argument must be a number. Given: {} and other: {}'.format(_func_name, self, other))
+
+
+@overload_method(SeriesType, 'unique')
+def hpat_pandas_series_unique(self, sorted=False):
+    """
+    Pandas Series method :meth:`pandas.Series.unique` implementation.
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_unique
+
+    Parameters
+    -----------
+    self: :class:`pandas.Series`
+        input arg
+    sorted: :obj:`bool`
+        return array is sorted
+
+    Returns
+    -------
+    :obj:`numpy.array`
+         returns :obj:`numpy.array` ndarray
+    """
+
+    _func_name = 'Method unique().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError(
+            '{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    def hpat_pandas_series_unique_impl_sorted(self):
+        return np.unique(self._data)
+
+    def hpat_pandas_series_unique_impl(self):
+        unique_values = []
+        for value in self._data:
+            if value not in unique_values:
+                unique_values.append(value)
+        return unique_values
+
+    return hpat_pandas_series_unique_impl_sorted if sorted else hpat_pandas_series_unique_impl
```

Comment:
I think you could control parameter `soret` inside `_impl` function by `return_index` parameter to `np`
```
>>> import numpy as np
>>> a = [4,2,1,3,1,2,3,4]
>>> np.unique(a)
array([1, 2, 3, 4])
>>> indexes = np.unique(a, return_index=True)[1]
>>> [a[index] for index in sorted(indexes)]
[4, 2, 1, 3]
```
https://stackoverflow.com/questions/12926898/numpy-unique-without-sort

Reply:
I'm afraid Numba doesn't support return_index parameter inside unique() method:
`TypeError: np_unique() got an unexpected keyword argument 'return_index'`